### PR TITLE
perf: various optimizations from Nasr

### PR DESCRIPTION
**Pull request by ivelin**
Clean PR porting code from @sniperadmin 's #365 and merging it with the latest code in master.
Also added a few e2e tests.

### CODE REVIEW

**Comment by ivelin** [comment]
@sniperadmin Can you take a look at this PR. I tried to merge your code from #365.

**Comment by ivelin** [comment]
commit-lint is complaining that the subject is longer than 100 characters. I can shorten that in the merge. Minor issue.

File: .github/workflows/lighthouse.yml [hunk 1]
```diff
@@ -35,29 +35,28 @@ jobs:
...
-    #     # Path to a LHCI lighthouserc.json file
-    #     # Opt-out of saving Lighthouse results as an action artifacts
-    #     uploadArtifacts: false
-    #     # Number of runs to do per URL
-    #     serverBaseUrl: ${{ secrets.LHCI_SERVER_URL }}
-    #     serverToken: ${{ secrets.LHCI_SERVER_TOKEN }}
-    #   env:
-    #     DEPLOY_URL: ${{ steps.waitFor200.outputs.url }}
-    
+    # runs the lighthouse action (deployed)
```
Comment (sniperadmin) [inline .github/workflows/lighthouse.yml]: I think this config will be overridden later, right?

Comment (ivelin) [reply .github/workflows/lighthouse.yml]: Your work on lhci and netflify testing was merged into master with [PR 367](https://github.com/ambianic/ambianic-ui/pull/367) a few days ago. 

This PR only pulls in your work on vuetify component optimization. Custom amb components and a few new cypress tests.

Comment (ivelin) [reply .github/workflows/lighthouse.yml]: BTW, github labeled the file you are commenting on as outdated. It is not the version in this PR branch.

**Comment by ivelin** [comment]
Looks like Cypress, NYC and the rest of the test stack are [doing a nice job measuring coverage](https://github.com/ambianic/ambianic-ui/pull/371/checks?check_run_id=687235818#step:5:2785) on js and vue files. We are at 42%! Yay. 50% is on the horizon.

File: .github/workflows/lighthouse.yml (hunk 1 above)
Comment (sniperadmin) [reply .github/workflows/lighthouse.yml]: oh i see now

Comment (sniperadmin) [reply .github/workflows/lighthouse.yml]: oh i see now

**Comment by sniperadmin** [comment]
im happy with the current results 😄

**Comment by ivelin** [comment]
OK, I will merge this in. Thank you for taking a look.

Is there anything else from #365 that needs to be merged? Or can we close it now?

**Pull request merged**